python_backend/generate_pptx.py

```
@@ -145,20 +154,82 @@ def __init__(self, original_file_path: str):
         """URLの場合はファイルをダウンロード"""
         if file_path.startswith(('http://', 'https://')):
+            masked_url, host, path_prefix = self._mask_url(file_path)
+            start_time = time.time()
+
             try:
-                logger.info(f"Downloading file from URL: {file_path}")
+                logger.debug(f"event=download_start host={host} path_prefix={path_prefix} timeout_s={HTTP_TIMEOUT_SECONDS}")
                 with tempfile.NamedTemporaryFile(suffix='.pptx', delete=False) as tmp_file:
-                    with urllib.request.urlopen(file_path) as response:
-                        tmp_file.write(response.read())
+                    with urllib.request.urlopen(file_path, timeout=HTTP_TIMEOUT_SECONDS) as response:
```

Ruff (0.14.11)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)